core/trino-spi/src/main/java/io/trino/spi/connector/RetryMode.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+public enum RetryMode
+{
+    NONE(false),
+    QUERIES(true),
+    TASKS(true);
```

Comment:
It may matter I think. I would expect that for some connectors you may implement query retries more efficiently. Then some connector-side transaction needs to be created per each query-attempt. While for TASK-level retries you would need to do that for each task-attempt (and there can be much more of those).

Reply:
> I would expect that for some connectors you may implement query retries more efficiently.

do we have a concrete example already? maybe this never happens. we could defer this complexity until we need it, could we?

Reply:
I would expect this to be true for JDBC connectors. For query-based retries we could have simpler logic there. But if we go for implementing two versions of logic; for tasks and queries; IDK.
I can drop it but then it would be natural to have boolean instead enum (only two values). And enum is nicer to read :P

Also I think it does not add too much complexity to SPI, so I am leaning towards keeping it.

Reply:
Chatted more about this one with @findepi . And realized that we do not have lifecycle information through query processing which would allow us to distinguish TASK/QUERY while individual splits are processed, and during query commit. And given fact that we probably do not want to invest heavily in QUERY retries, I will change the SPI so we do not distinguish QUERIES from TASKS here.